sdk/python/feast/client.py

```diff
@@ -470,35 +473,51 @@ def get_online_features(
         )  # type: GetOnlineFeaturesResponse
 
     def ingest(
-        self,
-        feature_set: Union[str, FeatureSet],
-        source: Union[pd.DataFrame, str],
-        version: int = None,
-        force_update: bool = False,
-        max_workers: int = CPU_COUNT,
-        disable_progress_bar: bool = False,
-        chunk_size: int = 5000,
-        timeout: int = None,
-    ):
+            self,
+            feature_set: Union[str, FeatureSet],
+            source: Union[pd.DataFrame, str],
+            chunk_size: int = 10000,
+            version: int = None,
+            force_update: bool = False,
+            max_workers: int = CPU_COUNT - 1,
```

Comment:
What if I have 1 core?

Reply:
My rationale for doing a `CPU_COUNT - 1` was to account for the process that was being used by the main thread.

I failed to account for the case where the Python SDK is running on a 1 core system. My apologise on that.

I will change this to min(CPU_COUNT - 1, 1)

Reply:
> My rationale for doing a `CPU_COUNT - 1` was to account for the process that was being used by the main thread.
> 
> I failed to account for the case where the Python SDK is running on a 1 core system. My apologise on that.
> 
> I will change this to `min(CPU_COUNT - 1, 1)`
> 
> In such a case, the main thread will only spawn a pool with 1 worker. (Total of 2 processes under the scope of Feast's Python SDK)

Ok. `min` or `max`?

Reply:
:| `max` was what i meant.

Reply:
fixed